### ISSUE TITLE
Debug as > Error while launching client file

### ISSUE THREAD

**Issue by Starsk44** [angelozerr/angularjs-eclipse#181]
Hi,

First thanks for you amazing work!!

In Eclipse the Run As > Protractor works fine !!!! Thanks for that!!

I have an issue regarding the Debug As > Protractor.

I have installed : http://oss.opensagres.fr/angularjs-eclipse/1.2.0-SNAPSHOT/
I have installed Webclipse JavaScript Debugger (http://marketplace.eclipse.org/marketplace-client-intro?mpc_install=2456312)

regarding the section : https://github.com/angelozerr/angularjs-eclipse/wiki/Protractor#debug

1/ in the preference page of "Angular>Protractor", in the "Run/Debug protactor with" I have no "Webeclipse Javascript Debugger" in the list... 
![image](https://cloud.githubusercontent.com/assets/16803759/12457665/1f456516-bfa6-11e5-9801-56b013b547d5.png)

2/ In the preference page of "JavaScript>Tern>Server>Node.js" in the "Run in debug mode with:" the same, I have no choice ", no "Webeclipse Javascript Debugger" in the list... 
![image](https://cloud.githubusercontent.com/assets/16803759/12458071/234914c6-bfa8-11e5-875f-67821121c6f1.png)

=> *Did I forget something during the install ?*

Regards, 
Arnaud

**Comment by angelozerr** [comment]
> First thanks for you amazing work!!

Thanks!

> Did I forget something during the install ?

No it was a mistake. You must install "Tern Debuggers". I have fixed the update site:

![terndebuggersupdatesite](https://cloud.githubusercontent.com/assets/1932211/12463725/9761ecd0-bfc5-11e5-87f7-0a1c860ab42b.png)

Please install "Tern Debuggers".

**Comment by Starsk44** [comment]
> You must install "Tern Debuggers". I have fixed the update site:

Nice!

> Please install "Tern Debuggers".

Done ;)

Now, I have one more Issue ("Selected Tern Repository is not imported as project in your workspace").
What I have done:

1/ In "Preferences > AngularJS > Protractor" I can now select the "Webclipse Javascript Debugger" Excellent!!
![image](https://cloud.githubusercontent.com/assets/16803759/12476745/85ca03c0-c02b-11e5-8cf8-87cc806d3ea3.png)

2/ In "Preferencies > JavaScript > Tern > Server Node.js"  
![image](https://cloud.githubusercontent.com/assets/16803759/12476889/a7fc2fb2-c02c-11e5-855e-85e41176b232.png)

* I have checked "Run in debug mode with" and have selected "Webclipse Javascript Debugger".
* "with Tern repository" I have no choice, so I have selected "default <not imported>" (maybe not a good idea). When "default <not imported>" is selected, I have the following Error : * "Selected Tern Repository is not imported as project in your workspace." *

Did I forget something ? Any Advices ?

Regards,
Arnaud

**Comment by angelozerr** [comment]
Tern debugger is used to debug any node application. Today we can debug:

 * Protractor with Debug As launch.
 * Tern plugins: when you do completion, hover, validation, etc inside JS Editor you can debug ternjs and tern plugins. So in your case you need not this feature (you wish to debug Protractor and not tern plugins). To debug tern plugins, you must import ternjs in your workspace (the goal of Import button is to import ternjs in your workspace).

In other words, in your case you need not to debug ternjs. As you have installed Webclipse tern debugger + Webclipse the "Debug As" Protractor should work.

**Comment by Starsk44** [comment]
First of all, thank you a lot for your fast response and your great support !!

> In other words, in your case you need not to debug ternjs. As you have installed Webclipse tern debugger + Webclipse the "Debug As" Protractor should work.

When I run "Debug As > Protractor" I have the following error :
![image](https://cloud.githubusercontent.com/assets/16803759/12478061/322a0816-c033-11e5-84de-f95f908c876b.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478091/60a16b1c-c033-11e5-95a0-4849c7ce063f.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478109/7e99ff76-c033-11e5-9753-c043a25fb17d.png)
org.eclipse.core.expressions
Error
Thu Jan 21 11:35:56 CET 2016
No property tester contributes a property org.chromium.debug.core.isJsFile to type class org.eclipse.core.internal.resources.File

Have you ever faced a situation similar to the case described above? 


My conf (General Properties) :  
![image](https://cloud.githubusercontent.com/assets/16803759/12478147/9f11d33c-c033-11e5-8fe8-116be31f0807.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478194/d950dfb6-c033-11e5-8ce0-246708053183.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478221/f5a5aa20-c033-11e5-9d15-26f61500b93d.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478231/03f5d780-c034-11e5-89c8-f3d9f5405a07.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478238/0da4520c-c034-11e5-874a-71a86c2b977b.png)

My conf (Project properties):  
![image](https://cloud.githubusercontent.com/assets/16803759/12478256/2ee17dc8-c034-11e5-9c7f-1d10aaff3a81.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478264/4266a6fc-c034-11e5-9c65-befa20356e70.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12478271/53aec642-c034-11e5-89e6-4fff47b13d02.png)

Installed Software:
![image](https://cloud.githubusercontent.com/assets/16803759/12478319/983f7856-c034-11e5-90c2-81c3f293352d.png)

**Comment by angelozerr** [comment]
It's an issue with Webclipse. @piotrtomiak  have you an idea?

**Comment by piotrtomiak** [comment]
@angelozerr indeed

@Starsk44 can you see if normal Webclipse debugging with Node.js works? Just create a simple "test.js" file with one line ```console.log("test")``` and right click on it -> Debug As -> Standalone Node.js Application.

It might be an issue with installation, some conflict with nodeclipse features? Can you show me the list of installed plugins? Type ```org.chromium``` in the filter field and ```javascript```.

**Comment by piotrtomiak** [comment]
@angelozerr Hmm... For me it keeps saying that it "Cannot find client file. Do you want to update Protractor preferences". I have imported everything as described in the doc. Maybe it has something to do with linux or spaces in the path?

**Comment by angelozerr** [comment]
@piotrtomiak when yiu will have time, could you set a breakpoint at https://github.com/angelozerr/angularjs-eclipse/blob/master/org.eclipse.angularjs.ui/src/org/eclipse/angularjs/internal/ui/launchConfigurations/ProtractorLaunchShortcut.java#L159 ?

**Comment by piotrtomiak** [comment]
@angelozerr Ok, so the issue is fixed. What I've found regarding the issue reported by @Starsk44 is that you are reusing existing launch configuration without updating it's configuration. So if a user launches protractor for config.js with external tool, it will than launch always with external tool. And it is the external tool which fails to launch in debug mode. I have the opposite situation in my workspace, I've successfully launched Protractor with node and now I am unable to launch it with external tool.

**Comment by Starsk44** [comment]
Yo,

FYI : I have found a workaround :    

Menu > Debug configurations ...
Create a new "Standalone Javascrip Application"

![image](https://cloud.githubusercontent.com/assets/16803759/12480413/6c85f42a-c042-11e5-8d07-47ae889bfedc.png)

![image](https://cloud.githubusercontent.com/assets/16803759/12480429/914647ec-c042-11e5-96fe-9a5862242244.png)

run this new debug conf ... and it works !!!!
![image](https://cloud.githubusercontent.com/assets/16803759/12480492/ff763d12-c042-11e5-9c08-8945bb2e8713.png)

**Comment by piotrtomiak** [comment]
@Starsk44 glad to hear that it works for you! I think that workaround is not enough though :) Let's keep it open until we get the Protractor launch issue figured out.

**Comment by piotrtomiak** [comment]
@angelozerr my proposition is to create a normal launch configuration, which would appear in the Debug/Run dialogs and could be edited over there. That launch config would spawn nodejs/nodeclipse/cli launches. Also, you need to "override" launch mode to "run" for cli "debug" launches, as the external tools launch delegate cannot handle "debug" mode.

**Comment by Starsk44** [comment]
>  I think that workaround is not enough though :) Let's keep it open until we get the Protractor launch issue figured out.

Ok Ok, let me know If I can help you here.

Thank you guys!! Thanks a lot for your amazing work and support.
Have a good day!

Cheers 
Arnaud

**Comment by piotrtomiak** [comment]
@angelozerr I've create PRs to solve all the reported issues. As for the launch configuration persistence, proposed fix is temporary just to keep people working with existing code. Once we have a normal Protractor launch configs, we will be able to reuse launches. We are now in a very hot period as we work on MyEclipse release on Mars, but after that I will be able to help you/write the Protractor launch configuration.
